src/content/docs/apm/agents/go-agent/instrumentation/go-agent-code-level-metrics-instrument.mdx

```diff
@@ -0,0 +1,124 @@
+---
+title: Go agent code-level metrics instrumentation
+tags:
+  - Agents
+  - Go agent
+  - Instrumentation
+metaDescription: Add source code context to your application traces, associating the trace data with the specific code being instrumented.
+---
+
+After you've turned on code-level metrics using the steps in [Go agent code-level metrics configuration](content/docs/apm/agents/go-agent/configuration/go-agent-code-level-metrics-config), you can fine-tune how your metrics are collected by using additional instrumentation. For general information about code-level metrics, see [Performance monitoring with CodeStream](/docs/codestream/how-use-codestream/performance-monitoring#code-level).
+
+## Instrumenting Transactions [#instrument-transactions-clm]
+
+With code-level metrics enabled, the Go Agent will add source code context information to any transaction started by a call
+to the `StartTransaction` method of the application. Without changing any of your existing instrumentation code, this means
+that the source code location will be added based on the function which called `StartTransaction`. This is true even if
+`StartTransaction` is called for you by an instrumentation package.
+
+However, if you wish to exert some manual control over how code-level metrics are collected for a particular transaction,
+you may add a number of `newrelic.TraceOption` functions as described below:
+
+### Suppressing Code-Level Metrics for a Single Transaction [#instrument-transactions-suppress-clm]
+If you don't want code-level metrics for a particular transaction (say, to avoid the overhead of collecting the information when you know you won't need that data),
+add a `WithoutCodeLevelMetrics` function to the end of the `StartTransaction` call:
+```go
+txn := app.StartTransaction("nothing-here-to-see", newrelic.WithoutCodeLevelMetrics())
+defer txn.End()
+```
+
+### Adjusting the File Path Prefix [#instrument-transactions-path-prefix-clm]
+We saw in the [configuration guide](/docs/apm/agents/go-agent/configuration/go-agent-code-level-metrics-config)
+that we can guide the agent's heuristic to skip functions it believes are internal to the agent itself
+to accurately find the one that started the transaction by setting the `ConfigCodeLevelMetricsIgnorePrefix` value.
+You can customize this on an individual transaction basis by adding a `WithIgnorePrefix` option to the
+end of the parameters to `StartTransaction`.
+```go
+txn := app.StartTransaction("mytransaction", newrelic.WithIgnorePrefix("github.com/ignore/this/"))
+defer txn.End()
+```
+
+### Explicitly Mark the Code Location [#instrument-transactions-this-location-clm]
+By adding `WithThisCodeLocation` to the end of the `StartTransaction` call, you can compensate for any difficulty that
+may occur in identifying the point of interest to be reported (for example, if the transaction is actually started
+inside another framework). This will force the code-level metrics to report the
```

Comment:
Same here! This sentence drags on for a while and it's confusing. Can you cut words out--like "compensate" "difficulty" "occur" "identifying?"

Reply:
@homelessbirds OK, I completely revamped this sentence. I'll confirm with the SME that the meaning is still the same.